Microsoft.Data.Analysis.Interactive/DataFrameKernelExtension.cs

```diff
@@ -0,0 +1,131 @@
+                var footer = new List<IHtmlContent>();
+                BuildHideRowsScript(uniqueId);
+
+                if (df.Rows.Count > SIZE)
+                {
+                    var paginateScriptFirst = BuildHideRowsScript(uniqueId) + GotoPageIndex(uniqueId, 0) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptFirst]("⏮"));
+
+                    var paginateScriptPrevTen = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, -10, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptPrevTen]("⏪"));
+
+                    var paginateScriptPrev = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, -1, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptPrev]("◀️"));
+
+                    footer.Add(b[style: "margin: 2px;"]("Page"));
+                    footer.Add(b[id: $"page_{uniqueId}", style: "margin: 2px;"]("1"));
+
+                    var paginateScriptNext = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, 1, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptNext]("▶️"));
+
+                    var paginateScriptNextTen = BuildHideRowsScript(uniqueId) + UpdatePageIndex(uniqueId, 10, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptNextTen]("⏩"));
+
+                    var paginateScriptLast = BuildHideRowsScript(uniqueId) + GotoPageIndex(uniqueId, (maxRows - 1) / SIZE) + BuildPageScript(uniqueId, SIZE);
+                    footer.Add(button[style: "margin: 2px;", onclick: paginateScriptLast]("⏭️"));
+                }
+                else
+                {
+                    BuildHideRowsScript(uniqueId);
+                    footer.Add(b[style: "margin: 2px;"]("Page"));
+                    footer.Add(b[id: $"page_{uniqueId}", style: "margin: 2px;"]("0"));
+                }
+
+                //table
+                var t = table[id: $"table_{uniqueId}"](
+                    caption(title),
+                    thead(tr(header)),
+                    tbody(rows.Select(r => tr[style: "display: none"](r))),
+                    tfoot(tr(td[colspan: df.Columns.Count + 1, style: "text-align: center;"](footer)))
+                );
+                writer.Write(t);
+
+                //show first page
+                writer.Write($"<script>{BuildPageScript(uniqueId, SIZE)}</script>");
+
+            }, "text/html");
+
+            return Task.CompletedTask;
+
+
+            static string BuildHideRowsScript(long uniqueId)
+            {
+                var script = $"var allRows = document.querySelectorAll('#table_{uniqueId} tbody tr:nth-child(n)'); ";
```

Comment:
I wonder if there is a more maintainable way of writing this JS code, like a resource or something.
@jonsequitur @colombod ?

Reply:
Not that I know of .. :(